### PR TITLE
Feature/hab 14 sign in with apple

### PR DESCRIPTION
**Pull request by HKurylchykava**

### CODE REVIEW

File: BadHabbit/BadHabbit/UserStories/SignInScreen/SignInViewController.swift [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+//  BadHabbit
+//
+//  Created by Halina Kurylchykava on 17.06.23.
+//
+
+import UIKit
+import AuthenticationServices
+
+class SignInViewController: UIViewController {
+    let userIdentifierLabel = UILabel()
```
Comment (nikGayko) [inline BadHabbit/BadHabbit/UserStories/SignInScreen/SignInViewController.swift]: Don't forget about marking private what shouldn't be used outside of this class

File: BadHabbit/BadHabbit/UserStories/SignInScreen/SignInViewController.swift [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            let userIdentifier = appleIDCredential.user
+            let fullName = appleIDCredential.fullName
+            let email = appleIDCredential.email
+            showUserData(userIdentifier: userIdentifier, givenName: fullName?.givenName, familyName: fullName?.familyName, email: email)
+        case let passwordCredential as ASPasswordCredential:
+            let username = passwordCredential.user
+            let password = passwordCredential.password
+
+            DispatchQueue.main.async {
+                self.showPasswordCredentialAlert(username: username, password: password)
```
Comment (nikGayko) [inline BadHabbit/BadHabbit/UserStories/SignInScreen/SignInViewController.swift]: why this call from main queue, but `showUserData` called earlier not from main?

File: BadHabbit/BadHabbit/UserStories/SignInScreen/SignInViewController.swift (hunk 1 above)
Comment (HKurylchykava) [reply BadHabbit/BadHabbit/UserStories/SignInScreen/SignInViewController.swift]: changed

File: BadHabbit/BadHabbit/UserStories/SignInScreen/SignInViewController.swift (hunk 2 above)
Comment (HKurylchykava) [reply BadHabbit/BadHabbit/UserStories/SignInScreen/SignInViewController.swift]: changed

**Pull request closed without merging**